src/mmu/timers.rs

```diff
@@ -0,0 +1,292 @@
+    fn get_system_clock_bit_for_tima(&self) -> u8 {
+        let byte = self.read_byte(TAC_ADDR);
+        let value = byte & 0b_0000_0011;
+        match value {
+            0 => TAC_FREQ_0_SYSTEM_CLOCK_BIT,
+            3 => TAC_FREQ_3_SYSTEM_CLOCK_BIT,
+            2 => TAC_FREQ_2_SYSTEM_CLOCK_BIT,
+            1 => TAC_FREQ_1_SYSTEM_CLOCK_BIT,
+            _ => unreachable!("Impossible value for TAC clock select"),
+        }
+    }
+
+    fn get_tima_bit_state(&self, prev_state: bool) -> bool {
+        let initial_tima_system_clock_bit = self.get_system_clock_bit_for_tima();
+        let system_clock = if prev_state {
+            self.timers.system_clock_prev
+        } else {
+            self.timers.system_clock
+        };
+        (system_clock & (1 << initial_tima_system_clock_bit)) != 0
+    }
+
+    // ----- Special-Case Memory Reads/Writes -----
+
+    /// Writes to the DIV register do not directly change its value. Instead, they reset
+    /// whe system clock. However, resetting the system clock can also increment TIMA
+    /// if it unsets TIMA's active bit!
+    pub fn write_byte_div(&mut self) {
+        let tima_bit_was_active = self.get_tima_bit_state(false);
+
+        self.timers.system_clock = 0;
+
+        let tima_bit_is_active = self.get_tima_bit_state(false);
+        if tima_bit_was_active && !tima_bit_is_active {
+            self.timers.tima_falling_edge_detected = true;
+        }
+    }
+
+    /// Writing to TIMA immediately after it has overflowed will cancel the overflow behavior.
+    /// Additionally, for the next machine cycle after TIMA overflows, it ignores writes!
+    pub fn write_byte_tima(&mut self, byte: u8) {
+        if self.timers.tima_write_lock_counter == 0 {
+            self.write_byte_override(TIMA_ADDR, byte);
+        }
+        self.timers.tima_overflowed = false;
+    }
+
+    /// While TIMA is write locked, writing to TMA will also write to TIMA (bypassing the lock).
+    /// Additionally, writing to TMA can cause timer ticks in TIMA.
+    pub fn write_byte_tma(&mut self, byte: u8) {
+        if self.timers.tima_write_lock_counter != 0 {
+            self.write_byte_override(TMA_ADDR, byte);
+        }
+    }
+
+    /// If TIMA was tracking a set bit, but a change to TAC changes its tracked bit to an unset bit,
+    /// TIMA will see that is a falling edge and increment.
+    /// Additionally, disabling the timer while TIMA's selected bit was set will
+    /// also trigger an increment
+    pub fn write_byte_tac(&mut self, byte: u8) {
+        let tima_bit_was_active = self.get_tima_bit_state(false);
+
+        self.write_byte_override(TAC_ADDR, byte);
+
+        let tima_bit_is_active = self.get_tima_bit_state(false);
+        let tima_is_enabled = self.get_tac_enable();
+
+        if tima_bit_was_active && (!tima_bit_is_active || !tima_is_enabled) {
+            self.timers.tima_falling_edge_detected = true;
+        }
+    }
+}
+
+mod debug {
```

Comment:
I'm not sure why this would be preferable to implementing the Debug trait on Mmu.  If there is some value add, add a comment. Otherwise, use the std library traits so all tools can take advantage of the debugging support.